src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesAuthFactory.java

```diff
@@ -0,0 +1,106 @@
+                )
+        );
+    }
+
+    public static KubernetesAuth fromCredentialsId(String credentialsId, String serviceAddress, String caCertData, Boolean skipTlsVerify) throws Exception {
+        StandardCredentials c = getAuthenticationTokenCredentials(credentialsId);
+        if (c != null) {
+            return AuthenticationTokens.convert(KubernetesAuth.class, c);
+        }
+        c = getCredentials(credentialsId);
+        if (c instanceof StringCredentials) {
+            return new KubernetesAuthToken(((StringCredentials) c).getSecret().getPlainText());
+        } else if (c instanceof TokenProducer) {
+            return new KubernetesAuthToken(((TokenProducer) c).getToken(serviceAddress, caCertData, skipTlsVerify));
+        } else if (c instanceof UsernamePasswordCredentials) {
+            UsernamePasswordCredentials usernamePassword = (UsernamePasswordCredentials) c;
+            return new KubernetesAuthUsernamePassword(
+                    usernamePassword.getUsername(),
+                    usernamePassword.getPassword().getPlainText()
+            );
+        } else if (c instanceof StandardCertificateCredentials) {
+            StandardCertificateCredentials certificateCredentials = (StandardCertificateCredentials) c;
+            KeyStore keyStore = certificateCredentials.getKeyStore();
+            String password = certificateCredentials.getPassword().getPlainText();
+            String alias = keyStore.aliases().nextElement();
+            X509Certificate certificate = (X509Certificate) keyStore.getCertificate(alias);
+            return new KubernetesAuthCertificate(
+                    Utils.wrapWithMarker(
+                        Utils.BEGIN_CERTIFICATE,
+                        Utils.END_CERTIFICATE,
+                        Base64.encodeBase64String(
+                            certificate.getEncoded()
+                        )
+                    ),
+                    Utils.wrapWithMarker(
+                        Utils.BEGIN_PRIVATE_KEY,
+                        Utils.END_PRIVATE_KEY,
+                        Base64.encodeBase64String(
+                            keyStore.getKey(alias, password.toCharArray()).getEncoded()
+                        )
+                    ),
+                    password
+            );
+        } else if (c instanceof DockerServerCredentials) {
+            DockerServerCredentials certificateCredentials = (DockerServerCredentials) c;
+            return new KubernetesAuthCertificate(certificateCredentials.getClientCertificate(), certificateCredentials.getClientKey());
+        } else if (c instanceof FileCredentials) {
+            try (InputStream is = ((FileCredentials) c).getContent()) {
+                return new KubernetesAuthKubeconfig(IOUtils.toString(is, StandardCharsets.UTF_8));
+            }
+        } else {
+            throw new Exception("Unable to use " + credentialsId + " for authentication");
```

Comment:
Create your own exception type or use something like `IllegalArgumentException`. In any case, if the credentials can't be handled, this is probably a programming error (or someone messed with the credentialsId manually) and it should just be logged.